contrib/images/babylond/Dockerfile

```diff
@@ -2,57 +2,38 @@ FROM golang:1.21 AS build-env
 
 # Version to build. Default is empty
 ARG VERSION
-
 ARG BUILD_TAGS=""
 ARG LEDGER_ENABLED="false"
-# Cosmos build options
 ARG COSMOS_BUILD_OPTIONS=""
 
-# Install cli tools for building and final image
-RUN apt-get update && apt-get install -y make git bash gcc curl jq
-
-# Build
 WORKDIR /go/src/github.com/babylonlabs-io/babylon
-# First cache dependencies
-COPY go.mod go.sum /go/src/github.com/babylonlabs-io/babylon/
-RUN go mod download
-# Then copy everything else
 COPY ./ /go/src/github.com/babylonlabs-io/babylon/
-# If version is set, then checkout this version
+
+# Handle if version is set
 RUN if [ -n "${VERSION}" ]; then \
-        git fetch --all; \
+        git fetch origin tag ${VERSION} --no-tags; \
         git checkout -f ${VERSION}; \
     fi
 
+# Cache mod dependencies
+RUN go mod download
+
 RUN LEDGER_ENABLED=$LEDGER_ENABLED \
     BUILD_TAGS=$BUILD_TAGS \
     COSMOS_BUILD_OPTIONS=$COSMOS_BUILD_OPTIONS \
     LINK_STATICALLY=false \
     make build
 
-FROM debian:bookworm-slim AS run
-# Create a user
-RUN addgroup --gid 1137 --system babylon && adduser --uid 1137 --gid 1137 --system --home /home/babylon babylon
-RUN apt-get update && apt-get install -y bash curl jq wget
+FROM debian:bookworm-slim AS wasm-link
+ARG WASMVM_VERSION="v2.0.1"
+
+RUN apt-get update && apt-get install -y wget bash
 
 # Label should match your github repo
 LABEL org.opencontainers.image.source="https://github.com/babylonlabs-io/babylond:${VERSION}"
 
-# Install libraries
 # Cosmwasm - Download correct libwasmvm version
-COPY --from=build-env /go/src/github.com/babylonlabs-io/babylon/go.mod /tmp
-RUN WASMVM_VERSION=$(grep github.com/CosmWasm/wasmvm /tmp/go.mod | cut -d' ' -f2) && \
-    wget https://github.com/CosmWasm/wasmvm/releases/download/$WASMVM_VERSION/libwasmvm.$(uname -m).so \
-        -O /lib/libwasmvm.$(uname -m).so && \
-    # verify checksum
-    wget https://github.com/CosmWasm/wasmvm/releases/download/$WASMVM_VERSION/checksums.txt -O /tmp/checksums.txt && \
-    sha256sum /lib/libwasmvm.$(uname -m).so | grep $(cat /tmp/checksums.txt | grep libwasmvm.$(uname -m) | cut -d ' ' -f 1)
-RUN rm -f /tmp/go.mod
+RUN wget https://github.com/CosmWasm/wasmvm/releases/download/$WASMVM_VERSION/libwasmvm.x86_64.so \
```

Comment:
why can 't we leave the old scripts which automatically determines the environment and verify checksum ?

Reply:
we can, I can rollback those changes

Reply:
done at https://github.com/babylonlabs-io/babylon/pull/16/commits/0943bc662ab88e407c6bd930f005499dd1335287